numpy/core/arrayprint.py

```diff
@@ -285,7 +268,27 @@ def _get_format_function(data, precision, suppress_small, formatter):
             if key in fkeys:
                 formatdict[key] = formatter[key]
 
+    return formatdict
+
+def _get_format_function(data, precision, suppress_small, formatter):
+    """
+    find the right formatting function for the dtype_
+    """
+    dtype_ = data.dtype
+    if dtype_.fields is not None:
+        format_functions = []
+        for field_name in dtype_.names:
+            field_values = data[field_name]
+            is_array_field = 1 < field_values.ndim
+            format_function = _get_format_function(
+                    ravel(field_values), precision, suppress_small, formatter)
+            if is_array_field:
```

Comment:
I would prefer `if dtype_[field_name].shape != ():`, so we don't need `is_array_field`.